core/src/main/java/com/sequenceiq/cloudbreak/core/flow/ClusterBootstrapperErrorHandler.java

```diff
@@ -0,0 +1,109 @@
+
+    @javax.annotation.Resource
+    private Map<CloudPlatform, CloudPlatformConnector> cloudPlatformConnectors;
+
+    public void terminateFailedNodes(ContainerOrchestrator orchestrator, Stack stack, Set<Node> nodes) throws CloudbreakOrchestratorException {
+        InstanceGroup gateway = stack.getGatewayInstanceGroup();
+        InstanceMetaData gatewayInstance = gateway.getInstanceMetaData().iterator().next();
+        List<String> allAvailableNode = orchestrator.getAllAvailableNode(gatewayInstance.getPublicIp(), nodes);
+        List<Node> missingNodes = selectMissingNodes(nodes, allAvailableNode);
+        if (missingNodes.size() > 0) {
+            LOGGER.info(String.format("Bootstrap failed on %s nodes. These nodes will be terminated.", missingNodes.size()));
+            for (Node missingNode : missingNodes) {
+                InstanceMetaData hostInStack =
+                        instanceMetaDataRepository.findByPrivateAndPublicAddress(stack.getId(), missingNode.getPrivateIp(), missingNode.getPublicIp());
+                instanceMetaDataRepository.delete(hostInStack.getId());
+                LOGGER.info(String.format("InstanceMetadata deleted with %s id and %s name.", hostInStack.getId(), hostInStack.getInstanceId()));
+                Resource resource = resourceRepository.findByStackIdAndName(stack.getId(), hostInStack.getInstanceId());
+                InstanceGroup ig = instanceGroupRepository.findOneByGroupNameInStack(stack.getId(), hostInStack.getInstanceGroup().getGroupName());
+                ig.setNodeCount(ig.getNodeCount() - 1);
+                if (ig.getNodeCount() < 1) {
+                    throw new CloudbreakOrchestratorException(String.format("%s instancegroup nodecount was lower than 1 cluster creation failed.",
+                            ig.getGroupName()));
+                }
+                instanceGroupRepository.save(ig);
+                LOGGER.info(String.format("Decreased nodecount on %s instancegroup.", ig.getGroupName()));
+                deleteResourceandDependencies(stack, hostInStack);
+                if (resourceRepository.findOne(resource.getId()) != null) {
+                    resourceRepository.delete(resource.getId());
+                }
+            }
+        }
+    }
+
+    private List<Node> selectMissingNodes(Set<Node> clusterNodes, List<String> availableNodes) {
+        List<Node> missingNodes = new ArrayList<>();
+        for (Node node : clusterNodes) {
+            boolean contains = false;
+            for (String nodeAddress : availableNodes) {
+                if (nodeAddress.equals(node.getPrivateIp())) {
+                    contains = true;
+                    break;
+                }
+            }
+            if (!contains) {
+                missingNodes.add(node);
+            }
+        }
+        return missingNodes;
+    }
+
+    private void deleteResourceandDependencies(Stack stack, InstanceMetaData hostInStack) {
```

Comment:
please rename hostInStack to instance, and make the "a" uppercase in the method name